shivamsouravjha/stock-backend
# Setup Sentry

`sentrygin` was used to introduce gin-specific middleware to set tra. More [can be read here](https://docs.sentry.io/platforms/go/guides/gin/#usage).

### How to test
- Add `SENTRY_DSN` to `.env`
- Introduce panic, for example open `controllers/file_controller.go`
```
func a(span *sentry.Span) {
	span.Status = sentry.SpanStatusFailedPrecondition
	err := errors.New("Sample Error")
	sentry.CaptureException(err)

	panic("error in a")
}

func (f *fileController) ParseXLSXFile(ctx *gin.Context) {
  ... sentry code
  a()
}
```
- start server `go run main.go`
- trigger panic with CURL
```
curl -X POST http://localhost:4000/api/uploadXlsx   -F "files=dsp-tax-saver-fund.xlsx"
```
- See in log Sentry sents events
- <img width="1090" alt="Screenshot 2024-10-04 at 5 39 18 PM" src="https://github.com/user-attachments/assets/77ca8cb8-3b4c-4a77-bfc9-daafe17f86bd">
- Go to Sentry
- Verify details for panic are there
- <img width="1728" alt="Screenshot 2024-10-08 at 4 11 08 PM" src="https://github.com/user-attachments/assets/caf7ed35-f21d-45a9-b48b-b37dbf73099b">

- **Result:** issue should be visible with all the context
